toolchain/check/testdata/impl/lookup/min_prelude/final_placement.carbon

```diff
@@ -68,3 +68,29 @@ import library "interface_z";
 
 // TODO: Can not write a final impl on `D` outside the file where `D` is defined.
 final impl D as Z(()) where .X = () {}
+
+// --- todo_final_blanket_and_specialized_impl_in_same_file_as_interface.carbon
+
+interface Z(T:! type) {
+  let X:! type;
+}
+
+// Blanket final impl on Z in the same file as Z.
+final impl forall [T:! type] T as Z(T) where .X = {.z_t: ()} {}
+
+class C;
+
+// final impl specialized on C for Z, also in the same file as Z.
+//
+// TODO: It's not fully decided yet, but at the moment, specializations of final
+// impls are not chosen when there's a more general impl available, even when it
+// is written in the same file.
+final impl C as Z(C) where .X = {.z_c: ()} {}
```

Comment:
If this impl is not prioritized, it is subsumed, which would be an error. Perhaps change the prioritization so they are overlapping but not subsets?

Reply:
I was actually going for that - this was the "`final A as B(?)` meaning as attached to the `B`" case from https://discord.com/channels/655572317891461132/941071822756143115/1369014619149963345 but let me make that more clear and add both meanings.

Reply:
Adjusted things and added a couple more tests PTAL